src/raftstore/coprocessor/mod.rs

```diff
@@ -102,6 +102,11 @@ pub trait SplitChecker {
 
     /// Get the desired split keys.
     fn split_key(&mut self) -> Option<Vec<u8>>;
+
+    /// Get approximate split keys without scan.
+    fn approximate_split_key(&self, _: &Region, _: &DB) -> Result<Option<Vec<u8>>> {
```

Comment:
Can you combine this into `split_key`? We don't need to know how it gets the split key outside.

Reply:
since to get not approximate split key, we must call on_kv() first. So outside caller it does know what the policy is.